airflow/providers/cncf/kubernetes/utils/delete_from.py

```diff
@@ -0,0 +1,141 @@
+                    body=body,
+                    **kwargs,
+                )
+            except client.rest.ApiException as api_exception:
+                api_exceptions.append(api_exception)
+    else:
+
+        try:
+            _delete_from_yaml_single_item(
+                k8s_client,
+                yml_document,
+                verbose,
+                namespace=namespace,
+                body=body,
+                **kwargs,
+            )
+        except client.rest.ApiException as api_exception:
+            api_exceptions.append(api_exception)
+
+    if api_exceptions:
+        raise FailToDeleteError(api_exceptions)
+
+
+def _delete_from_yaml_single_item(
+    k8s_client,
+    yml_document,
+    verbose=False,
+    namespace="default",
+    body=None,
+    **kwargs,
+):
+    if body is None:
+        body = DEFAULT_DELETION_BODY
+
+    # get group and version from apiVersion
+    group, _, version = yml_document["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    # Only replace the last instance
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split("."))
+    fcn_to_call = f"{group}{version.capitalize()}Api"
+    k8s_api = getattr(client, fcn_to_call)(k8s_client)
+    # Replace CamelCased action_type into snake_case
+    kind = yml_document["kind"]
+    kind = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", kind)
+    kind = re.sub("([a-z0-9])([A-Z])", r"\1_\2", kind).lower()
+
+    # Decide which namespace we are going to use for deleting the object
+    # IMPORTANT: Its ignore namespace in args:
+    #    create_from_yaml_single_item have same behaviour
+    if "namespace" in yml_document["metadata"]:
+        namespace = yml_document["metadata"]["namespace"]
+    name = yml_document["metadata"]["name"]
+
+    # Expect the user to delete namespaced objects more often
+    if hasattr(k8s_api, f"delete_namespaced_{kind}"):
+        resp: client.V1Status = getattr(k8s_api, f"delete_namespaced_{kind}")(
+            name=name, namespace=namespace, body=body, **kwargs
+        )
+    else:
+        resp: client.V1Status = getattr(k8s_api, f"delete_{kind}")(name=name, body=body, **kwargs)
+    if verbose:
+        print(f"{kind} deleted. status='{str(resp.status)}'")
+    return resp
+
+
+class FailToDeleteError(Exception):
+    """
+    An exception class for handling error if an error occurred when
+    handling a yaml file during deletion of the resource.
+    """
+
+    def __init__(self, api_exceptions):
+        self.api_exceptions = api_exceptions
+
+    def __str__(self):
+        msg = ""
+        for api_exception in self.api_exceptions:
+            msg += f"Error from server ({api_exception.reason}):{api_exception.body}"
```

Comment:
Won't these all run together?

Reply:
if there is a list of multiple resource, each deletion is try.

if an error occur then it will fail at the end and log the error

Reply:
More specifically, does the exception body have a newline at the end? Or will these just get strung together into a single line with no space/delimiter?

Reply:
I added a `\n`